packages/web/src/executors/rollup/rollup.impl.ts

```diff
@@ -1,327 +1,21 @@
-        use: {
-          less: {
-            javascriptEnabled: options.javascriptEnabled,
-          },
-        },
-      }),
-      resolve({
-        preferBuiltins: true,
-        extensions: fileExtensions,
-      }),
-      useSwc && swc(),
-      useBabel &&
-        getBabelInputPlugin({
-          // Let's `@nrwl/web/babel` preset know that we are packaging.
-          caller: {
-            // @ts-ignore
-            // Ignoring type checks for caller since we have custom attributes
-            isNxPackage: true,
-            // Always target esnext and let rollup handle cjs/umd
-            supportsStaticESM: true,
-            isModern: true,
-          },
-          cwd: join(context.root, sourceRoot),
-          rootMode: 'upward',
-          babelrc: true,
-          extensions: fileExtensions,
-          babelHelpers: 'bundled',
-          skipPreflightCheck: true, // pre-flight check may yield false positives and also slows down the build
-          exclude: /node_modules/,
-          plugins: [
-            format === 'esm'
-              ? undefined
-              : require.resolve('babel-plugin-transform-async-to-promises'),
-          ].filter(Boolean),
-        }),
-      commonjs(),
-      analyze(),
-    ];
-
-    const globals = options.globals
-      ? options.globals.reduce(
-          (acc, item) => {
-            acc[item.moduleId] = item.global;
-            return acc;
-          },
-          { 'react/jsx-runtime': 'jsxRuntime' }
-        )
-      : { 'react/jsx-runtime': 'jsxRuntime' };
-
-    const externalPackages = dependencies
-      .map((d) => d.name)
-      .concat(options.external || [])
-      .concat(Object.keys(packageJson.dependencies || {}));
-
-    const rollupConfig = {
-      input: options.entryFile,
-      output: {
-        globals,
-        format,
-        dir: `${options.outputPath}`,
-        name: options.umdName || names(context.projectName).className,
-        entryFileNames: `[name].${format === 'esm' ? 'js' : 'cjs'}`,
-        chunkFileNames: `[name].${format === 'esm' ? 'js' : 'cjs'}`,
-        // umd doesn't support code-split bundles
-        inlineDynamicImports: format === 'umd',
-      },
-      external: (id) =>
-        externalPackages.some(
-          (name) => id === name || id.startsWith(`${name}/`)
-        ) || npmDeps.some((name) => id === name || id.startsWith(`${name}/`)), // Could be a deep import
-      plugins,
-    };
-
-    return options.rollupConfig.reduce((currentConfig, plugin) => {
-      return require(plugin)(currentConfig, options);
-    }, rollupConfig);
-  });
-}
-
-function createCompilerOptions(options, dependencies) {
-  const compilerOptionPaths = computeCompilerOptionsPaths(
-    options.tsConfig,
-    dependencies
+  rollupExecutor as _rollupExecutor,
+  RollupExecutorOptions as WebRollupOptions,
+} from '@nrwl/rollup';
+
+export { WebRollupOptions };
+
+export async function* rollupExecutor(options: any, context: any) {
```

Comment:
nitpick: let's give these parameters better types?

Reply:
Going to remove this later and don't want to set a direct dependency on `@nrwl/rollup` moving forward.